pandas/core/groupby.py

```diff
@@ -2859,7 +2859,7 @@ def is_in_obj(gpr):
         else:
             in_axis, name = False, None
 
-        if is_categorical_dtype(gpr) and len(gpr) != len(obj):
+        if is_categorical_dtype(gpr) and len(gpr) != obj.shape[axis]:
             raise ValueError("Categorical dtype grouper must "
                              "have len(grouper) == len(data)")
```

Comment:
In light of the change in the check, perhaps this error message should change.

Reply:
Good point. I'll change it to "Grouper and axis must be same length".

Reply:
put in the actual len(grouper) and len(axis) as well